app.js incrementado

RFPadilha opened the pull request.

RFPadilha committed:
app.js incrementado

bruno-braga commented on workAround.js:
```diff
@@ -0,0 +1,144 @@
+let character = {//usado de exemplo
```
You can remove the comments here as the variable describes itself very well. In addition, if any comment is necessary let's use english instead of pt-br.

bruno-braga commented on workAround.js:
```diff
@@ -0,0 +1,144 @@
+}
+
+
+//DEFINIÇÕES DE ATRIBUTOS
```
Remove this comment, also the variable describes itself well.

bruno-braga commented on workAround.js:
```diff
@@ -0,0 +1,144 @@
+let methodChoice=0;//se 0 usa standardArray, se 1 recebe input do usuario
+var atributoAuxiliar;
+
+//while(methodChoice!=1 || methodChoice!=0){ //comentado pq não funciona como deveria, de resto, recebe valores do usuario like a champ
```
It is not necessary to commit code that is commented as it is not being used. Commented code is noisy and adds complexity to the code reader.

bruno-braga commented on workAround.js:
```diff
@@ -0,0 +1,144 @@
+  console.log(Math.floor((sheet.dexterity-10)/2))
+}
+
+//DEFINIÇÕES DE SKILLS
```
Remove this comment, also the variable describes itself well. The variable can be removed as well as it is not being used.

bruno-braga commented on workAround.js:
```diff
@@ -0,0 +1,144 @@
+      sheet[atributes[i]] = isAtributeValid;
+      usedArray = remove(usedArray, usedArray.indexOf(isAtributeValid));
+
+    }else i--;//deselegante as fuck
```
Remove this comment, as it doesn't add any value to the code.

bruno-braga commented on workAround.js:
```diff
@@ -0,0 +1,144 @@
+var atributoAuxiliar;
+
+//while(methodChoice!=1 || methodChoice!=0){ //comentado pq não funciona como deveria, de resto, recebe valores do usuario like a champ
+	methodChoice = Number(prompt("Qual o método de atribuição que será utilizado?\n 0-Standard Array, 1-Inserção manual"))
```
methodChoice doesn't describe itself very well. Here in order to rename this variable try to think about what value is being store in the variable or even the action that creates the value.

bruno-braga commented on workAround.js:
```diff
@@ -0,0 +1,144 @@
+let atributes = ["strength", "dexterity", "constitution", "intelligence", "wisdom", "charisma"]
+let usedArray = [];
+let methodChoice=0;//se 0 usa standardArray, se 1 recebe input do usuario
+var atributoAuxiliar;
```
This variable is written in pt-br. Change to es. Also is there any specific reason for using var here? Can be change to let?

bruno-braga commented on workAround.js:
```diff
@@ -0,0 +1,144 @@
+
+let standardArray = [15, 14, 13, 12, 10, 8];
+let atributes = ["strength", "dexterity", "constitution", "intelligence", "wisdom", "charisma"]
+let usedArray = [];
```
Rename this variable to something that describes the value that the variable holds.

bruno-braga commented on workAround.js:
```diff
@@ -0,0 +1,144 @@
+
+
+
+function remove(array, index) {
```
This function can be renamed as well. The name should describe what the function does. Try to rename it by using this statement as reference: 

remove a value of the array by index.

pdrlaguna replied:
Concordo 100%, mas não tava no escopo da feature/desse PR (aí a importância de nomear bem as branchs e os commits de forma semântica, sem "patch 1", "another change", "fix typo") é essa. O escopo era criar o "mod" de um stat - não era refatorar o código pra ficar mais legível, não era skills... Se o nome da branch fosse "calculating stat modifier" a gente talvez não estaria discutindo sobre como o stat é construído.

Nosso processo tá bem atravessado nesse sentido ainda.

bruno-braga commented on workAround.js:
```diff
@@ -0,0 +1,144 @@
+
+
+
+/*next steps:
```
This kind comments should be on the README on this repository as they describe how the sheet generator works.

bruno-braga reviewed (changes requested):
Things to notice:

1. Identation from line 26 to 38 is wrong. Probably because that block was scoped within the while loop that is commented(A very good example why commented code shoudn't be commited).
2. Git is being used here so we can check it out the differences the versions of our code. That's why here there is no necessity to create a second file. So all the changes being asked here you should fix onto app.js.

RFPadilha committed:
Rename app.js to apps.js

Added distribution method of random rolls and user input.
Added method of choosing skills to be proficient in
-Needs limiting through future class, race, background and feats implementations
-Needs proper wording on prompts

RFPadilha committed:
Delete apps.js

RFPadilha committed:
Added new features

Added method of random rolls and user input for stat distribution
-implemented mod calculator
Added character descriptors
-Implemented function to calculate level and proficiency through exp
Added method of skill choosing
-Needs proper prompt wording
-Needs limiting by future class/race/background/feats features

RFPadilha committed:
Update README.md

RFPadilha committed:
Update README.md

RFPadilha committed:
Update README.md
